### PR TITLE
fix(parser): API Gateway V2 request context scope field should be optional

### LINKED ISSUE #2953: Bug: RequestContextV2AuthorizerJwt requires scopes
### Expected Behaviour

I'm using the RequestContextV2 Model for parsing the event payload within a AWS Lambda function called by an AWS API Gateway. The API Gateway is configured to allow only authenticated requests with a valid JWT Token. The token is generated by our self-hosted GitLab instance with the help of the https://docs.gitlab.com/ee/ci/yaml/#id_tokens keyword. 

The token itself has no scopes and it's not possible to configure any. The API Gateway is configured without any scope requirements which is a valid configuration according to AWS docs.

### Current Behaviour

Right now this configuration is not supported by the RequestContextV2AuthorizerJwt Model as it requires that a scopes field is present in the object. The validation fails and an exception is thrown

[ERROR] ValidationError: 1 validation error for APIGatewayProxyEventV2Model
requestContext.authorizer.jwt.scopes
  Input should be a valid list [type=list_type, input_value=None, input_type=NoneType]
    For further information visit https://errors.pydantic.dev/2.1/v/list_type
Traceback (most recent call last):
  File "/opt/python/lib/python3.10/site-packages/aws_lambda_powertools/middleware_factory/factory.py", line 135, in wrapper
    response = middleware()
  File "/opt/python/lib/python3.10/site-packages/aws_lambda_powertools/utilities/parser/parser.py", line 83, in event_parser
    parsed_event = parse(event=event, model=model, envelope=envelope) if envelope else parse(event=event, model=model)
  File "/opt/python/lib/python3.10/site-packages/aws_lambda_powertools/utilities/parser/parser.py", line 155, in parse
    return envelope().parse(data=event, model=model)
  File "/opt/python/lib/python3.10/site-packages/aws_lambda_powertools/utilities/parser/envelopes/apigwv2.py", line 30, in parse
    parsed_envelope: APIGatewayProxyEventV2Model = APIGatewayProxyEventV2Model.parse_obj(data)
  File "/opt/python/lib/python3.10/site-packages/typing_extensions.py", line 2562, in wrapper
    return __arg(*args, **kwargs)
  File "/opt/python/lib/python3.10/site-packages/pydantic/main.py", line 961, in parse_obj
    return cls.model_validate(obj)
  File "/opt/python/lib/python3.10/site-packages/pydantic/main.py", line 496, in model_validate
    return cls.__pydantic_validator__.validate_python(

### Code snippet

```python
@event_parser(model=Pipeline, envelope=envelopes.ApiGatewayV2Envelope)
def lambda_handler(event: Pipeline, context: LambdaContext):
  pass
```

### Possible Solution

https://github.com/aws-powertools/powertools-lambda-python/compare/develop...smittnacht:powertools-lambda-python:jwt-scopes-as-optional

### Steps to Reproduce

Create an AWS API Gateway with JWT Authorizer and with a Lambda Function integration. Don't configure any required scopes. Send an authenticated request to the endpoint without any scopes defined.

### Powertools for AWS Lambda (Python) version

latest

### AWS Lambda function runtime

3.10

### Packaging format used

Lambda Layers

### Debugging logs

_No response_

### PR DESCRIPTION
**Pull request by leandrodamascena**
**Issue number:** #2953 

### Changes

Include the Optional type to make the "scope" field optional.

### User experience

> Please share what the user experience looks like before and after this change

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda/python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: aws_lambda_powertools/utilities/parser/models/apigwv2.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ class RequestContextV2AuthorizerIam(BaseModel):
 
 class RequestContextV2AuthorizerJwt(BaseModel):
     claims: Dict[str, Any]
-    scopes: List[str]
+    scopes: Optional[List[str]] = None
```
Comment (heitorlessa) [inline aws_lambda_powertools/utilities/parser/models/apigwv2.py:28]: can we have a quick test to avoid future regression?

We can take this event, remove the `scopes` dict key and run the model - no need to assert is `None`, if the Model works without it, it means validation worked as expected.

https://github.com/aws-powertools/powertools-lambda-python/blob/bfde1c9124910851f3de41b1db54e48e2bba7f16/tests/unit/parser/test_apigwv2.py#L41

**Review by heitorlessa — CHANGES_REQUESTED** [review]
tiny test to avoid regressions

File: aws_lambda_powertools/utilities/parser/models/apigwv2.py (hunk 1 above)
Comment (leandrodamascena) [reply aws_lambda_powertools/utilities/parser/models/apigwv2.py:28]: I would like to keep this test and this field to validate that we are parsing as a List when a value is provided.

My thoughts:

1 - I can add more lines in this test to remove this field and run the model again
```python
    .
    .
    .
    assert parsed_event.stageVariables == raw_event["stageVariables"]

    # Removing scope field
    raw_event["requestContext"]["authorizer"]["jwt"]["scopes"] = None
    parsed_event: APIGatewayProxyEventV2Model = APIGatewayProxyEventV2Model(**raw_event)
```

2 - I can create another test to test specific this, but I think this is too much in my opinion.

What do you suggest?

Comment (heitorlessa) [reply aws_lambda_powertools/utilities/parser/models/apigwv2.py:28]: Create another test specific to the claim being none. It'll be only a few lines, and if it fails we know what contract it broke.

I'll post a sample what I mean by that after a meeting

Comment (heitorlessa) [reply aws_lambda_powertools/utilities/parser/models/apigwv2.py:28]:
```python
def test_apigw_v2_event_empty_jwt_scopes():
    raw_event = load_event("apiGatewayProxyV2Event.json")
    raw_event["requestContext"]["authorizer"]["jwt"]["scopes"] = None

    APIGatewayProxyEventV2Model(**raw_event)
```

It fails (without this PR)

```python
```python
E   pydantic.error_wrappers.ValidationError: 1 validation error for APIGatewayProxyEventV2Model
E   requestContext -> authorizer -> jwt -> scopes
E     none is not an allowed value (type=type_error.none.not_allowed)

pydantic/main.py:341: ValidationError
```

```

Comment (leandrodamascena) [reply aws_lambda_powertools/utilities/parser/models/apigwv2.py:28]: Thanks for creating the test @heitorlessa! I created a new one before you send this haha! :heart:

**Review by heitorlessa — APPROVED** [review]
